### ISSUE TITLE
Search bar in History blocks buttons in UI

### ISSUE THREAD

**Issue by Brave-Matt** [brave/brave-browser#15572]
## Description 

When you search your browsing history in `Settings --> History`, the search bar appears and overlays the `Cancel` and `Delete` buttons you would use after selecting sites:

Without search:
![image](https://user-images.githubusercontent.com/3299098/116611535-56161200-a8eb-11eb-9f6b-9bc8c141f7e9.png)

With search:
![image](https://user-images.githubusercontent.com/3299098/116611709-89f13780-a8eb-11eb-8173-cec6d70b46d8.png)

## Steps to Reproduce

   1. Visit `Settings --> History`
   2. Search using the search bar at the top
   3. Observe buttons being hidden

## Reproduces how often: 

Always

## Brave version (brave://version info)

Brave | 1.24.77 Chromium: 90.0.4430.85 (Official Build) dev (x86_64)
-- | --
Revision | 5bc145d831c180d9ff94f29a0d7a2e1cbd30ef36-refs/branch-heads/4430@{#1311}
OS | macOS Version 11.2.3 (Build 20D91)

Note that this also happens on Windows systems.

## Version/Channel Information:

- Can you reproduce this issue with the current release? Yes
- Can you reproduce this issue with the beta channel? Yes
- Can you reproduce this issue with the nightly channel? Yes

## Miscellaneous Information:

User report:
https://community.brave.com/t/a-bug-in-ui-in-browser-history-page/240494

**Comment by rebron** [comment]
Working as designed with the `x` button for closing the search. So I'm not sure there's a need for cancel and delete buttons for search whereas I see the need for selection.
cc: @petemill @karenkliu

**Comment by karenkliu** [comment]
Yeah, this is actually how Chrome works. You can't search history after selecting an item in history (in Gmail, can you still use the search function if something is selected, but searching mail will get rid of your selection and instead show the search results). 
In our case, it looks funky because we have other stuff in the top nav that Chrome doesn't have, which also gets hidden if the user is engaging in searching history.

I propose leaving as is for now, but to make it less confusing in the future we might want to take the page-specific search icon out of the top nav - it seems that some people don't realize it's searching only the page and not ALL the content through Settings/Brave Rewards/History/Bookmarks/Downloads/Crypto Wallets. Also, some pages like Crypto Wallets don't require a page-search function. 

Dropping the search function down to the page level might look something like this:
![image](https://user-images.githubusercontent.com/2388823/116910537-9e3e7880-abfa-11eb-8689-6fb47be2c738.png)

Of course, we'll have to think of the UI treatment for each the pages (Settings/Brave Rewards/History/Bookmarks/Downloads/Crypto Wallets) if we go in this direction.

**Comment by eljuno** [comment]
Duplicate of #7398 ?

**Comment by rebron** [comment]
Closing as dupe (and stale). Fixed in 1.28.x
